sklearn/preprocessing/_label.py

```diff
@@ -112,6 +116,8 @@ def _encode(values, uniques=None, encode=False, check_unknown=True):
             res = _encode_python(values, uniques, encode)
         except TypeError:
             raise TypeError("argument must be a string or number")
```

Comment:
I think we should simply be changing this message to avoid "argument"

Reply:
You mean

`raise TypeError("must be a string or number")`

or just `raise TypeError`?

Reply:
Ideally something like `TypeError("Encoders require their input to be strings or numbers. Got {}")`